CRM/Civicase/Hook/PreProcess/CaseCategoryWordReplacementsForChangeCase.php

```diff
@@ -35,10 +36,45 @@ public function run($formName, CRM_Core_Form &$form) {
   private function addWordReplacements(CRM_Core_Form $form) {
     $caseCategoryName = CaseCategoryHelper::getCategoryName($form->_caseId[0]);
     CRM_Civicase_Hook_Helper_CaseTypeCategory::addWordReplacements($caseCategoryName);
-    // We need to translate this manually as Civi does not the page title
-    // through the ts function.
+  }
+
+  /**
+   * Sets the Page title.
+   *
+   * We need to translate this manually as Civi did not translate the page
+   * title. Because the form object has no function to get the current
+   * page title, we need to re-construct the page title similar to what is
+   * done in activity form class.
+   *
+   * @param CRM_Core_Form $form
+   *   Form Object.
+   */
+  private function setPageTitle(CRM_Core_Form $form) {
     $pageTitle = $form->get_template_vars('activityTypeName');
-    CRM_Utils_System::setTitle(ts($pageTitle));
+    $displayName = $this->getContactDisplayName($form);
+    if ($displayName) {
+      CRM_Utils_System::setTitle(ts($displayName . ' - ' . $pageTitle));
+    }
+    else {
+      CRM_Utils_System::setTitle(ts('%1 Activity', [1 => $pageTitle]));
+    }
+  }
+
+  /**
+   * Function to get currently viewed contact display name.
+   *
+   * @param CRM_Core_Form $form
+   *   Form object.
+   *
+   * @return string
+   *   Contact display name.
+   */
+  private function getContactDisplayName(CRM_Core_Form $form) {
+    if ($form->_currentlyViewedContactId) {
+      $displayName = CRM_Contact_BAO_Contact::displayName($form->_currentlyViewedContactId);
+
+      return $displayName;
+    }
```

Comment:
this doesn't always return a string, is that ok?

Reply:
Yeah it's fine but will update return type to reflect it can be null or string.